osrf/vrx-docker
# minimal working examples of competitor images

This PR adds a new directory to that contains 3 examples of Docker images that a competitor might start from when creating a submission.

To test, prepare each docker image, upload it to dockerhub, and then follow the instructions to score it as a submission. Here are detailed instructions for each of the 3 images:

## Preparation
* Use the `docker login` command to log in to the `virtualrobotx` acount. [See instructions here](https://github.com/osrf/vrx-private/blob/master/utils/phase2_pr_test.md).
* Set a tag name that is unique to you. For example:
  ```
  TAG=mmccarr
  ```

## Test `vrx_2019_simple`
1. Set the name of the image and year you will test:
    ```
    TEAM=vrx_2019_simple
    YEAR=2019
    ```
1. Change into the directory containing the Docker file:
    ```
    cd ~/vrx_ws/src/vrx-docker/mwes/${YEAR}/${TEAM}
    ```
    * We need to make a temporary minor tweak to the 2019 run script so we can observe some behavior using VRX 2022. Use a text editor to open `run_my_system.bash` and edit the last two lines to read:
        ```
        rostopic pub /wamv/thrusters/left_thrust_cmd std_msgs/Float32 -r ${RATE} -- ${CMD} &
        rostopic pub /wamv/thrusters/right_thrust_cmd std_msgs/Float32 -r ${RATE} -- ${CMD}        
        ```
1. Build the image and tag with virtualrobotx:
    ```
    docker build -t virtualrobotx/${TEAM}:${TAG} .
    ```
1. Push the image to dockerhub:
    ```
    docker push virtualrobotx/${TEAM}:${TAG}
    ```
1. Change to the `team_config` directory:
    ```
    cd ~/vrx_ws/src/vrx-docker/team_config
    ```
1. Create a test team for this image:
    ```
    mkdir ${TEAM}
    cp example_team_2019/* ${TEAM}
    ```
1. Edit the `dockerhub_image.txt` file to point to your dockerhub image:
    ```
    echo "virtualrobotx/${TEAM}:${TAG}" > ${TEAM}/dockerhub_image.txt
    ```
1. Follow the [usual testing instructions](https://github.com/osrf/vrx-private/blob/master/utils/phase2_pr_test.md) to make sure the WAMV runs on each of our 6 trials and moves forward in a straight line as expected.
 
## 2. Test vrx_2022_simple
* Set the name and year of the image you will test:
    ```
    TEAM=vrx_2022_simple
    YEAR=2022
    ```
* Follow the rest of the instructions above, but skip the part of step 2 that involves editing `run_my_system.bash` and  use `example_team` instead of `example_team_2019` in step 6.

## 3. Test vrx_2022_starter
* Set the name and year of the image you will test:
    ```
    TEAM=vrx_2022_starter
    YEAR=2022
    ```
* Change into the directory containing the Docker file:
    ```
    cd ~/vrx_ws/src/vrx-docker/mwes/${YEAR}/${TEAM}
    ```
* Make a temporary copy of the VRX repository in this directory:
    ```
    git clone https://github.com/osrf/vrx my_source
    ```
* Note that after you build the image, you'll need to move this cloned directory out of your workspace so it doesn't cause a conflict with your other `vrx` source:
  ```
  mv my_source ~/my_source
  ```
* Follow the rest of the instructions for `vrx_2019_simple`, but skip the part of step 2 that involves editing `run_my_system.bash` and  use `example_team` instead of `example_team_2019` in step 6.

## Clean up
Discard the edits you made to `~/vrx_ws/src/vrx-docker/mwes/${YEAR}/${TEAM}/run_my_system.bash`
